### PR TITLE
Add t3c-check-reload

### PR DESCRIPTION
**Pull request by rob05c**
Adds t3c restart determiner, per https://github.com/apache/trafficcontrol/blob/master/blueprints/ort-rewrite-unix-style.md

No docs, no changelog, no tests: still experimental. The plan is to make manpages for docs, we have yet to come up with the system for that. Likewise, because it's primarily CLI it really needs an integration test framework, which is still being developed.

- [x] This PR is not related to any other Issue

## Which Traffic Control components are affected by this PR?
- Traffic Ops ORT

## What is the best way to verify this PR?
Make changes to TO that require reloads or restarts and run `t3c`, verify reloads or restarts are performed as necessary.

## If this is a bug fix, what versions of Traffic Control are affected?
Not a bug fix.

## The following criteria are ALL met by this PR
- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

## Additional Information

### CODE REVIEW

**Review by ezelkow1 — APPROVED** [review]
Looks good, tested on my lab cache with staging and updated when appropriate and when not

File: cache-config/t3c-check/t3c-check.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	"os"
+	"path/filepath"
+	"syscall" // TODO change to x/unix ?
+
+	"github.com/pborman/getopt/v2"
+)
+
+var commands = map[string]struct{}{
+	"refs":   struct{}{},
+	"reload": struct{}{},
```
Comment (zrhoffman) [inline cache-config/t3c-check/t3c-check.go:33]: This can be just
```go
	"refs":   {},
	"reload": {},
```

**Review by zrhoffman — CHANGES_REQUESTED** [review]

File: cache-config/t3c-apply/config/config.go [hunk 2]
```diff
@@ -23,6 +23,7 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	"github.com/apache/trafficcontrol/cache-config/t3cutil"
 	"github.com/apache/trafficcontrol/lib/go-log"
 	"github.com/pborman/getopt/v2"
```
[the remark this answers is not in the document]
Comment (rob05c) [reply cache-config/t3c-apply/config/config.go]: Changed

File: cache-config/t3c-check/t3c-check.go (hunk 1 above)
Comment (rob05c) [reply cache-config/t3c-check/t3c-check.go]: Changed

**Review by zrhoffman — APPROVED** [review]
Look's good! The CI seems backed up for the apache org right now, but it passed on [your fork](https://github.com/rob05c/trafficcontrol/actions?query=branch:ort-restart-determiner). Merging...

**Review by zrhoffman — COMMENTED** [review]
There are still references to `t3c-verify` in some files.

https://github.com/apache/trafficcontrol/blob/2637e1c2eb91761002a482f7adeb95bbeb829869/cache-config/t3c-check-refs/README.md#L26
https://github.com/apache/trafficcontrol/blob/2637e1c2eb91761002a482f7adeb95bbeb829869/cache-config/t3c/README.md#L34
https://github.com/apache/trafficcontrol/blob/2637e1c2eb91761002a482f7adeb95bbeb829869/cache-config/t3c/t3c.go#L98

**Comment by rob05c** [comment]
> There are still references to t3c-verify in some files.
Fixed in https://github.com/apache/trafficcontrol/pull/5852

**Pull request merged**